CONTRIBUTING.md

```diff
@@ -0,0 +1,105 @@
+# Contributing
+
+**Table of Contents**
+1. [Welcome](#welcome) (2nd draft)
+2. [Philosophy](#philosophy) (TBD)
+3. [Reporting an Issue](#reporting-an-issue) (TBD)
+4. [Fixing an Issue (PRs)](#fixing-an-issue-prs) (2nd draft)
+5. [Setup and Run](#setup-and-run) (TBD)
+
+
+## Welcome
+Hi, and welcome to Highlight.js. We're a code-highlighting engine etc., but you probably know that. You want to contribute, and this document will be your guide.
+
+There are many ways to contribute to Highlight.js. Reporting issues is super useful. Feel free to work on an open issue. You can also help by writing [a plugin](https://highlightjs.readthedocs.io/en/latest/plugin-api.html), adding [a 3rd party grammar](https://github.com/highlightjs/highlight.js/blob/master/extra/3RD_PARTY_QUICK_START.md), or creating [a theme](https://highlightjs.readthedocs.io/en/latest/style-guide.html).
+
+In terms of getting help, GitHub issues are the primary method of communication. We're exploring the idea of creating a Gitter chat room.
+
+
+## Philosophy
+
+Highlight.js makes a strong effort to keep the core engine, it's usage, and it's feature set clean and simple.  We aim to support the happy path (the 90%), not the often innumerable edge cases (the last 10%).  When you're on the happy path everything "just works" with a single line of code.  We do our best to allow those familiar with JavaScript to easily handle the remaining 10% via small extensions or plug-ins.
+
+Feature requests are always welcome but many times the answer is:
+
+> That's a great idea, you should write a plug-in for that! It doesn't belong in the core library and it's probably something you could do with only a small amount of code.
+
+If you'd like to contribute a feature you should always first open an issue to discuss if the feature belongs in core vs a plug-in.  Often this is a great way to get tips for implemention as well or links to prior disussions with context, etc.
+
+Some bullet points:
+
+- Our [lack of line numbers is a feature](https://highlightjs.readthedocs.io/en/latest/line-numbers.html). The simple case isn't hard to add though with a little custom JavaScript.
```

Comment:
Found a reference to the google group on that page... Someone monitors it?
And shouldn't we point users to https://github.com/taufik-nurrohman/highlight.ln.js?

Reply:
We could very briefly, though I think we need another separate place for "Awesome HLJS" stuff in general.